internal/controller/pipelinerollout_controller.go

```diff
@@ -451,11 +456,11 @@ func (r *PipelineRolloutReconciler) processExistingPipeline(ctx context.Context,
 //
 // - as long as there's no other requirement to pause, set desiredPhase=Running
 func (r *PipelineRolloutReconciler) processExistingPipelineWithoutDataLoss(ctx context.Context, pipelineRollout *apiv1.PipelineRollout,
-	existingPipelineDef, newPipelineDef *kubernetes.GenericObject, pipelineNeedsToUpdate bool) error {
+	existingPipelineDef, newPipelineDef *kubernetes.GenericObject, pipelineNeedsToUpdate bool, upgradeStrategy usde.UpgradeStrategy) error {
 
 	numaLogger := logger.FromContext(ctx)
 
-	shouldBePaused, err := r.shouldBePaused(ctx, pipelineRollout, existingPipelineDef, newPipelineDef, pipelineNeedsToUpdate)
+	shouldBePaused, err := r.shouldBePaused(ctx, pipelineRollout, existingPipelineDef, newPipelineDef, pipelineNeedsToUpdate, upgradeStrategy)
```

Comment:
I'm thinking we don't need to pause the pipeline if we're using Progressive. So, we could conditionally call this only if using PPND

Reply:
Even if it's determined that pausing is an aspect of Progressive, it won't happen here I don't think.

Reply:
At this time, we do not have Progressive implementation, so PPND and Progressive should both pause the pipeline for now. This logic will need to be changed once Progressive is available.

Reply:
I think it's a little confusing as is, though, since I don't think we need to change `shouldBePaused()` at all. There are a bunch of TODOs that you added in there that should go over here instead. Right now nobody should be using Progressive strategy at all, so we can do whatever we want if they call that, such as issue an error, or just use apply.